torchtext/experimental/datasets/language_modeling.py

```diff
@@ -74,31 +75,19 @@ def _setup_datasets(dataset_name, tokenizer=None, root='.data', vocab=None,
 
     if not single_line and dataset_name != 'WikiText103':
         raise TypeError('single_line must be True except for WikiText103')
-    if dataset_name == 'WMTNewsCrawl':
-        train, = raw.DATASETS[dataset_name](root=root, data_select=('train',))
-        if single_line:
-            raw_data = {'train': [" ".join([txt for txt in train]), ]}
-        else:
-            raw_data = {'train': [txt for txt in train]}
-    else:
-        train, test, valid = raw.DATASETS[dataset_name](root=root, data_select=('train', 'test', 'valid'))
-        # Cache raw text iterable dataset
-        if single_line:
-            raw_data = {'train': [" ".join([txt for txt in train]), ],
```

Comment:
All of this was being processed, even if the given data ends up being discarded due to data_select